audit.go

```diff
@@ -0,0 +1,162 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-querystring/query"
+)
+
+const auditBaseURL = "/audit/records"
+
+// ListAuditRecordsOptions is the data structure used when calling the
+// ListAuditRecords API endpoint.
+type ListAuditRecordsOptions struct {
+	Actions              []string `url:"actions,omitempty,brackets"`
+	ActorID              string   `url:"actor_id,omitempty"`
+	ActorType            string   `url:"actor_type,omitempty"`
+	Cursor               string   `url:"cursor,omitempty"`
+	Limit                uint     `url:"limit,omitempty"`
+	MethodTruncatedToken string   `url:"method_truncated_token,omitempty"`
+	MethodType           string   `url:"method_type,omitempty"`
+	RootResourcesTypes   []string `url:"root_resources_types,omitempty,brackets"`
+	Since                string   `url:"since,omitempty"`
+	Until                string   `url:"until,omitempty"`
+}
+
+// ListAuditRecordsResponse is the response data received when calling the
+// ListAuditRecords API endpoint.
+type ListAuditRecordsResponse struct {
+	Records          []AuditRecord    `json:"records,omitempty"`
+	ResponseMetaData ResponseMetaData `json:"response_metadata,omitempty"`
+	Limit            uint             `json:"limit,omitempty"`
+	NextCursor       string           `json:"next_cursor,omitempty"`
+}
+
+// AuditRecord is a audit trail record that matches the query criteria.
+type AuditRecord struct {
+	ID               string           `json:"id,omitempty"`
+	Self             string           `json:"self,omitempty"`
+	ExecutionTime    string           `json:"execution_time,omitempty"`
+	ExecutionContext ExecutionContext `json:"execution_context,omitempty"`
+	Actors           []Actor          `json:"actors,omitempty"`
```

Comment:
Let's switch this to be `[]APIObject`, I'll explain why in my comment on the `Actor` type definition...